File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/PlayerPickupEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.Objects;

public final class PlayerPickupEvent extends AbstractCancellableEvent {

    private final CellPlayer player;
    private final String itemId;
    private final int count;

    public PlayerPickupEvent(
            CellPlayer player,
            String itemId,
            int count
    ) {
        this.player = Objects.requireNonNull(player, "player");
        this.itemId = Objects.requireNonNull(itemId, "itemId");
        this.count = count;
    }

    public CellPlayer player() {
        return player;
    }

    public String itemId() {
        return itemId;
    }

    public int count() {
        return count;
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/PlayerPickupEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer

public class PlayerPickupEvent(
    private val player: CellPlayer,
    private val itemId: String,
    private val count: Int
) : AbstractCancellableEvent() {

    public fun player(): CellPlayer = player

    public fun itemId(): String = itemId

    public fun count(): Int = count

}
